include/networkit/graph/GraphTools.hpp

```diff
@@ -406,6 +406,24 @@ void sortEdgesByWeight(Graph &G, bool decreasing = false);
  */
 std::vector<node> topologicalSort(const Graph &G);
 
+/**
+ * Given a  grah G, randomizes the weights in the graph. The weights are distributed in
+ * the range [0, 1] by default, unless a distribution is provided. This occurs in-place
+ * if the graph is already weighted, but creates a weighted copy before randomizing weights
+ * if it wasn't.
+ *
+ * @param   G           Directed input graph
+ * @param   distr       Random distribution
+ * @return              A graph with random edge weights
+ */
+template <class Distribution = std::uniform_real_distribution<edgeweight>>
+void randomizeWeights(Graph &G, Distribution distr = std::uniform_real_distribution<edgeweight>{
+                                    0, std::nexttoward(1.0, 2.0)}) {
+    if (!G.isWeighted())
+        G = toWeighted(G);
+    G.forEdges([&](node u, node v) { G.setWeight(u, v, distr(Aux::Random::getURNG())); });
```

Comment:
This can run in parallel, which is important for large graphs. However, the result will not be deterministic. To have deterministic results we should either use a fixed random seed for each edge or run the algorithm sequentially.

Reply:
Does the result not being deterministic actually matter here? None of the previous uses it is substituting bothered making it sequential or setting a seed for each edge.

Reply:
The previous uses were internal, this is now a public API so I think it matters.

Reply:
It seems `forEdges` was inherently sequential. Made so that it will run parallel by default with the option to run sequentially.

Reply:
Having a sequential flag is not really a great API. Call it `deterministic` instead or so. I'd also prefer if it was the last parameter.